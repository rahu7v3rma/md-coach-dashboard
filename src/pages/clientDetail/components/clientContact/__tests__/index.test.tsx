import { create } from 'react-test-renderer';

import ClientContact from '..';
import Envelop from 'src/assets/envelop.svg';
import Phone from 'src/assets/phone.svg';

describe('ClientContact', () => {
    it('ClientContact Snapshot', () => {
        const tree = create(<ClientContact />);
        expect(tree).toMatchSnapshot();
    });

    it('renders ClientContact component with all props', () => {
        const tree = create(
            <ClientContact phoneNumber="123456789" email="admin@gmail.com" />
        );
        let clientContactProps = tree.root.findByProps({
            phoneNumber: '123456789',
            email: 'admin@gmail.com'
        }).props;

        expect(clientContactProps).toBeTruthy();
    });

    it('renders ClientContact component with missing props', () => {
        const tree = create(<ClientContact />);
        expect(tree).toBeTruthy();
    });

    it('ClientContact phone and email icons are displayed', () => {
        const tree = create(<ClientContact />);
        const phoneIcon = tree.root.findByProps({ id: 'phoneIcon' }).props;
        const emailIcon = tree.root.findByProps({ id: 'emailIcon' }).props;
        expect(phoneIcon.src).toEqual(Phone);
        expect(emailIcon.src).toEqual(Envelop);
    });
});
