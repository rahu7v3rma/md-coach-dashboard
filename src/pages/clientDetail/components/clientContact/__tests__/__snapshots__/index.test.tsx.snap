// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientContact ClientContact Snapshot 1`] = `
<div
  className="sc-ewnqHT jLSBow"
>
  <div
    className="sc-fFGjHI exUEJr"
  >
    <img
      className="sc-dicizt hhLdfQ"
      id="phoneIcon"
      src={Object {}}
    />
    <p
      className="sc-gTRrQi eWXWTs"
      color="#271A51"
      data-testid="phone_number"
      fontSize={16}
      fontWeight="500"
    />
  </div>
  <div
    className="sc-fFGjHI exUEJr"
  >
    <img
      className="sc-dicizt hhLdfQ"
      id="emailIcon"
      src={Object {}}
    />
    <p
      className="sc-gTRrQi eWXWTs"
      color="#271A51"
      data-testid="email"
      fontSize={16}
      fontWeight="500"
    />
  </div>
  <div
    className="sc-fFGjHI exUEJr"
  >
    <button
      className="sc-bgqQcB hrunJP"
      color="#271A51"
      data-testid="ageButton"
      fontSize={14}
      id="ageButton"
    />
    <span
      className="sc-iVCKna jaFFvY"
    />
    <button
      className="sc-bgqQcB bNGpES"
      color="#271A51"
      data-testid="typeButton"
      fontSize={14}
      id="typeButton"
    />
  </div>
</div>
`;
